src/compiler/transformers/ts.ts

```diff
@@ -1889,7 +1889,22 @@ export function transformTypeScript(context: TransformationContext) {
     function transformEnumMemberDeclarationValue(member: EnumMember): Expression {
         const value = resolver.getConstantValue(member);
         if (value !== undefined) {
-            return typeof value === "string" ? factory.createStringLiteral(value) : factory.createNumericLiteral(value);
+            if (typeof value === "string") {
+                return factory.createStringLiteral(value);
+            }
+            if (Number.isNaN(value)) {
+                return resolver.isNameReferencingGlobalValueAtLocation("NaN", member)
+                    ? factory.createIdentifier("NaN")
+                    : factory.createBinaryExpression(factory.createNumericLiteral(0), SyntaxKind.SlashToken, factory.createNumericLiteral(0));
+            }
+            if (!isFinite(value)) {

```

Comment:
What do you think of this alternative way to write it?

```ts
if (!isFinite(value) {
    let result: Node = resolver.isNameReferencingGlobalValueAtLocation("Infinity", member) ?
        factory.createIdentifier("Infinity") :
        factory.createBinaryExpression(factory.createNumericLiteral(0), SyntaxKind.SlashToken, factory.createNumericLiteral(0));
    if (value < 0) {
        result = factory.createPrefixUnaryExpression(SyntaxKind.MinusToken, result);
    }
    return result;
}
```

Reply:
I guess that would emit as `-(1/0)`

Reply:
> What do you think of this alternative way to write it?

That's what I started with :p
https://github.com/microsoft/TypeScript/pull/55107/commits/ed2cc5717d0f956d90592af943141449b4b6f05d

> I guess that would emit as -(1/0)?

and why I changed it to the current version 😉